File: mobile/ios-native-starter/DiningDealzLiquidGlassHeaderButtonViewManager.m
#import <React/RCTViewManager.h>
#import <React/RCTComponent.h>

@interface RCT_EXTERN_REMAP_MODULE(DiningDealzLiquidGlassHeaderButtonView, DiningDealzLiquidGlassHeaderButtonViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(accessibilityLabel, NSString)
RCT_EXPORT_VIEW_PROPERTY(label, NSString)
RCT_EXPORT_VIEW_PROPERTY(systemImage, NSString)
RCT_EXPORT_VIEW_PROPERTY(variant, NSString)
RCT_EXPORT_VIEW_PROPERTY(onGlassButtonPress, RCTDirectEventBlock)

@end

File: mobile/ios/DiningDealzLiquidGlassHeaderButtonViewManager.swift
import Foundation
import React

@objc(DiningDealzLiquidGlassHeaderButtonViewManager)
final class DiningDealzLiquidGlassHeaderButtonViewManager: RCTViewManager {
  override func view() -> UIView! {
    DiningDealzLiquidGlassHeaderButtonView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    true
  }
}